extension/include/boost/di/extension/injections/shared_factory.hpp

```diff
@@ -17,7 +17,25 @@
 BOOST_DI_NAMESPACE_BEGIN
 namespace extension {
 
-template <class T, class TFunc>
+struct no_recursion {};
```

Comment:
Did you consider just use it as a fwd declaration? No need to instansiate that guy, right?

Reply:
Can you please elaborate a bit?

Reply:
struct no_recursion;